rust/datafusion/src/execution/table_impl.rs

```diff
@@ -131,6 +132,11 @@ impl Table for TableImpl {
     ) -> Result<Vec<RecordBatch>> {
         ctx.collect_plan(&self.plan.clone(), batch_size)
     }
+
+    /// Returns the schema from the logical plan
+    fn schema(&self) -> &Box<Schema> {
```

Comment:
I'm curious, would it be possible to just return `&Schema` here? I know we tend to return `Arc<Schema>` in many places and this is a pattern I introduced when I was less familiar with Rust, but it would be good to start moving away from this pattern where we can.

Reply:
So I was able to get the ref of the boxed value using `as_ref` -- let me know if that is what you were intending.